leader-election-impl/src/main/java/com/palantir/leader/lease/ClockReversalDetector.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.leader.lease;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Suppliers;
+
+final class ClockReversalDetector {
```

Comment:
tempted to remove this and instead put a reversal check in the actual token class

Reply:
so i'm actually just going to remove this and replace with an implementation of the clock skew monitor; but this can go into a FLUP PR.

Reply:
we will probably need to have a ReversalDetector on client side for the client lock leases. I can take care of removing this class in that PR.